lib/components/ModalPortal.js

```diff
@@ -153,29 +153,14 @@ export default class ModalPortal extends Component {
     }
   }
 
-  handleOverlayMouseDown = () => {
-    if (this.shouldClose === null) {
-      this.shouldClose = true;
-    }
-  }
-
-  handleOverlayMouseUp = (event) => {
-    if (this.shouldClose && this.props.shouldCloseOnOverlayClick) {
-      if (this.ownerHandlesClose()) {
-        this.requestClose(event);
-      } else {
-        this.focusContent();
-      }
+  handleOverlayClick(event) {
+    if (this.props.shouldCloseOnOverlayClick && this.ownerHandlesClose()) {
+      this.requestClose(event);
     }
-    this.shouldClose = null;
-  }
-
-  handleContentMouseDown = () => {
-    this.shouldClose = false;
   }
 
-  handleContentMouseUp = () => {
-    this.shouldClose = false;
+  handleContentClick(event) {
```

Comment:
Also...

Lint: `static handleContentClick(event)`.

[eslint/class-methods-use-this](http://eslint.org/docs/rules/class-methods-use-this)
> If a class method does not use this, it can safely be made a static function.

Hope this one is correct.